src/field/captcha/third_party_captcha.jsx

```diff
@@ -166,6 +209,8 @@ export class ThirdPartyCaptcha extends React.Component {
   }
 
   componentDidMount() {
+    // grab the render params outside of the callback just to spy on it in the test
+    this.getRenderParams();
```

Comment:
I don't think we should have this comment here.

Additionally, I am not sure I understand why we need `this.getRenderParams` in the first place, we shouldnt have to spy on it. Instead, we should verify what gets passed to `provider.render` as that's what counts. `getRenderParams` is an implementation details we shouldnt test, meaning that if we revert this to the original, tests should still succeed.

Reply:
Thanks for making all the changes here, appreciate it.

Currently, the code relies on `this.getRenderParams` to have been called on mount, and then just uses `this.renderParams`.

How do you feel about dropping `this.renderParams`, but instead call `this.getRenderParams()` where u need the params, and inline the current baseRendrParams in the getRenderParams. It would also remove the need to do `delete this.renderPrams`, which I always try to avoid.

```js
 getRenderParams() {
    if (this.props.provider === ARKOSE_PROVIDER) {
      return;
    }
    
    const defaultRenderParams = {
      sitekey: this.props.sitekey,
    };

    if (this.props.provider === FRIENDLY_CAPTCHA_PROVIDER) {
      return {
        ...this.defaultRenderParams,
        language: this.props.hl,
        doneCallback: this.changeHandler,
        errorCallback: this.erroredHandler
      };
    }
    return {
      ...this.renderParams,
      callback: this.changeHandler,
      'expired-callback': this.expiredHandler,
      'error-callback': this.erroredHandler
    };

    if (this.props.provider === AUTH0_V2_CAPTCHA_PROVIDER) {
      return {
        ...this.defaultRenderParams,
        callback: this.changeHandler,
        'expired-callback': this.expiredHandler,
        'error-callback': this.erroredHandler
        language: this.props.hl,
        theme: 'light'
      };
    } else {
      return {
      ...this.defaultRenderParams,
      callback: this.changeHandler,
      'expired-callback': this.expiredHandler,
      'error-callback': this.erroredHandler
    };
    }
  }
```